### PR TITLE
add prefix check before update properties

### PR DESCRIPTION
**Pull request by nianma**
I don't have time to create new type exception. Also I would prefer some design this first like naming...

### CODE REVIEW

**Comment by nianma** [comment]
https://jira.duraspace.org/browse/FCREPO-1766

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 1]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
 
         final Collection<IllegalArgumentException> errors = checkInvalidPredicates(request);
 
+        request.getPrefixMapping().getNsPrefixMap().forEach(
+                       (k,v) -> { try {
+                                   LOGGER.info("451: prefix mapping is key -> value: " + k + "->" + v + "\n");
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Comments on your logging:
* When using slf4j (which you are), you should take advantage of templating. For example, the above line should be:
`LOGGER.info("451: prefix mapping is key -> value: {} -> {}", k, v);`
* This should be `debug` not `info`

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 2]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
...
+                  if ( Arrays.asList(getSession().getWorkspace().getNamespaceRegistry().getPrefixes()).contains(k)
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: The list of namespace prefixes should likely be created once, before this stream and reused.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 3]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
...
+                     && !v.equals(getSession().getWorkspace().getNamespaceRegistry().getURI(k))) {
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Likewise, you could probably retrieve the namespace-registry before this stream, and reuse it here.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 4]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
...
+                                     LOGGER.info("********451: has exists\n");
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Please remove this logging statement.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 5]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
...
+                                    throw new IllegalArgumentException( "prefix has existed");
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: Per @ajs6f's earlier comment, please create a new Exception type for this scenario.
For examples see: https://github.com/fcrepo4/fcrepo4/tree/master/fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/exception

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java [hunk 6]
```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
...
+
+                                   }
+                                  } catch (RepositoryException e) {
+                                    LOGGER.info("********catch " + e);
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: What is this error case? Do we really want to be swallowing the exception?

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java (hunk 1 above)
Comment (ajs6f) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java]: That better not be a reference to [HTTP 451](https://en.wikipedia.org/wiki/HTTP_451).

**Comment by awoods** [comment]
Please create an integration test that demonstrates the issue before your functional updates, and proves the fix after your functional updates.
You can probably add a test method to:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/test/java/org/fcrepo/integration/kernel/modeshape/FedoraResourceImplIT.java

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/FedoraInvalidPrefixExceptionMapper.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import javax.ws.rs.ext.Provider;
+
+import org.fcrepo.kernel.api.exception.FedoraInvalidPrefixException;
+
+import org.slf4j.Logger;
+
+/**
+ * For invalid namespace exceptions on CRUD actions for nodes/datastreams
+ *
+ * @author whikloj
```
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/FedoraInvalidPrefixExceptionMapper.java]: `@author` should be `nianma`

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/805712449d8677a218faa490901a449fec3011e0

**Pull request closed without merging**